File: 品味Java的21種設計模式/dp_cpp/factorymethod/example3/ExportTxtFileOperate.cpp
#include "ExportTxtFileOperate.h"

namespace cn
{
	namespace javass
	{
		namespace dp
		{
			namespace factorymethod
			{
				namespace example3
				{

					ExportFileApi *ExportTxtFileOperate::factoryMethod()
					{
						//创建导出成文本文件格式的对象
						return new ExportTxtFile();
					}
				}
			}
		}
	}
}

File: 品味Java的21種設計模式/dp_java/factorymethod/example3/ExportTxtFileOperate.java
package cn.javass.dp.factorymethod.example3;
/**
 * 具体的创建器实现对象，实现创建导出成文本文件格式的对象
 */
public class ExportTxtFileOperate extends ExportOperate {

    protected ExportFileApi factoryMethod() {
        //创建导出成文本文件格式的对象
        return new ExportTxtFile();
    }

}
